Further support for more carts

lifehackerhansol opened the pull request:
TLDR:
- ~~Added `dsbize.c` (unfortunately needed for the M3 family)~~
- Added support for:
    - ~~M3 DS Real and derivatives~~
    - ~~M3i Zero (GMP-Z003)~~
    - EZ-Flash V
    - EZ-Flash Parallel (lol)
    - Games 'n' Music
    - `r4dspro.com`
        - For now this is a placeholder name, it refers to Acekard 2 derivatives that does not have a functional SDIO CMD25. It's done this way as I didn't want to gimp write speed for all Acekards just to unify this one guy.
        - There's only really two different models with this quirk I've seen in the wild, but the r4dspro is still being sold as of this writing.
    - EDGEi
        - Not to be confused with the iEDGE, it's just the usual AK2i derivative
    - Stargate 3DS (lol)

lifehackerhansol committed:
Add support for the Datel Games 'n' Music (GMTF)

lifehackerhansol committed:
Add support for the EZ-Flash V (non-SDHC)

lifehackerhansol committed:
Add support for the EDGEi (Acekard 2i clone)

lifehackerhansol committed:
Add support for Acekard 2 derivatives with broken CMD25

- These are currently referred to as carts from "r4dspro.com", not too
  happy about the naming but it's at least recognizable by end users
- We can't just use the Acekard default driver as the card firmware
  can't do multi-block writes *at all*, and it would suck to gimp the
  write speed of other Acekards just for this one quirk

lifehackerhansol committed:
Add support for the EZ-Flash Parallel

Aside from the usual copy-and-DLDI-patch, we also need to remove the
DLDI magic number. The reason for this is that EZ-Flash's firmware is
just devkitPro's bootstrap, and thus it will auto-DLDI-patch our
application on the fly if the DLDI magic number is found.

Thus, we sed it to null.

lifehackerhansol committed:
Add support for the Stargate 3DS

asiekierka commented:
Can we split off the non-`dsbize` and `dsbize` carts into separate PRs? I'd still like to try and get rid of the "blobby" nature of replacing the whole header as `dsbize` does, if possible.

lifehackerhansol commented:
> Can we split off the non-`dsbize` and `dsbize` carts into separate PRs? I'd still like to try and get rid of the "blobby" nature of replacing the whole header as `dsbize` does, if possible.

Removed M3 devices from this PR.

asiekierka commented:
Thank you!
